### DISCUSSION TITLE
rust-course/关于 Self 的疑问

### DISCUSSION THREAD

**Discussion by AllenYu0118** [sunface/rust-course#392] [内容建议]
在 Rust 基础入门 - 2.8.4 章节 - 关联类型中：

> 还记得 Self 吧？在之前的章节[提到过](https://course.rs/basic/trait/trait-object#Self%E4%B8%8Eself)， Self 用来指代当前的特征实例，那么 Self::Item 就用来指代特征实例中具体的 Item 类型：

上文有提到 Self 用来指代当前特征实例，但是在 [提到过](https://course.rs/basic/trait/trait-object#Self%E4%B8%8Eself) 的章节中，说明的是 Self 指代的是类型，self 才是实例。

以上两种说法感觉有些冲突，该如何正确理解呢？

**Comment by sunface** [comment]
感谢建议。。。已经进行了相应的优化 :)
